File: appli-android/app/src/main/java/com/example/sportapp/core/network/ExerciseEquipmentApi.kt
package com.example.sportapp.core.network

import com.example.sportapp.core.data.model.ExerciseEquipment
import retrofit2.http.*

interface ExerciseEquipmentApi {

    @GET("exercise-equipments")
    suspend fun getAll(): List<ExerciseEquipment>

    @GET("exercise-equipments/{uuid}")
    suspend fun getByUUID(@Path("uuid") uuid: String): ExerciseEquipment



    @PUT("exercise-equipments/{uuid}")
    suspend fun upsert(@Path("uuid") uuid: String, @Body equipment: ExerciseEquipment)

    @PUT("exercise-equipments/bulk")
    suspend fun upsertAll(@Body equipments: List<ExerciseEquipment>)

    @DELETE("exercise-equipments/{uuid}")
    suspend fun delete(@Path("uuid") uuid: String)
}

File: appli-web/src/app/core/api/exercise-equipment-api.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { ExerciseEquipment } from '@core/models/exercise-equipment.model';
import { API_BASE_URL } from './api.config';

@Injectable({ providedIn: 'root' })
export class ExerciseEquipmentApi {
  private readonly http = inject(HttpClient);
  private readonly base = `${API_BASE_URL}/exercise-equipments`;

  getAll(): Observable<ExerciseEquipment[]> {
    return this.http.get<ExerciseEquipment[]>(this.base);
  }

  upsert(m: ExerciseEquipment): Observable<ExerciseEquipment> {
    // Corps wire attendu par ExerciseEquipmentCreate (camelCase).
    return this.http.put<ExerciseEquipment>(`${this.base}/${m.uuid}`, {
      uuid: m.uuid,
      exerciseUUID: m.exerciseUUID,
      equipmentUUID: m.equipmentUUID,
      updatedAt: m.updatedAt,
    });
  }

  /** Upsert groupé `PUT /exercise-equipments/bulk` (miroir upsertAll Android) — 1 requête pour N rows. */
  upsertAll(items: ExerciseEquipment[]): Observable<ExerciseEquipment[]> {
    return this.http.put<ExerciseEquipment[]>(
      `${this.base}/bulk`,
      items.map((m) => ({
        uuid: m.uuid,
        exerciseUUID: m.exerciseUUID,
        equipmentUUID: m.equipmentUUID,
        updatedAt: m.updatedAt,
      })),
    );
  }

  delete(uuid: string): Observable<unknown> {
    return this.http.delete(`${this.base}/${uuid}`);
  }
}
